template/src/app/types/@react-native-firebased.d.ts

```diff
@@ -0,0 +1,6 @@
+import {UserCredential} from 'firebase/auth';
```

Comment:
Hmm :thinking: the naming on this one `firebase**d**` ? Also, how does this interact with the other types from auth?

Reply:
Module Augmentation from typescript: https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation

Basically it merges those types to the existing ones.

Reply:
Fixed the typo and moved `types` folder to `src`

Reply:
cool - in fact module augmentation is how the whole firebase-js-sdk v8 (not v9!) type system worked
works for me :-)